driver/scsi_operation.c

```diff
@@ -547,6 +547,38 @@ WnbdModeSense(_In_ PWNBD_DISK_DEVICE Device,
     return SrbStatus;
 }
 
+UCHAR
+WnbdReportLuns(_In_ PWNBD_DISK_DEVICE Device,
+               _In_ PSCSI_REQUEST_BLOCK Srb) {
+    ASSERT(Srb);
+
+    PVOID DataBuffer = SrbGetDataBuffer(Srb);
+    ULONG DataTransferLength = SrbGetDataTransferLength(Srb);
+    UCHAR SrbStatus = SRB_STATUS_INTERNAL_ERROR;
+
+    if (NULL == DataBuffer) {
+        SrbStatus = SRB_STATUS_INTERNAL_ERROR;
+        goto Exit;
+    }
+
+    RtlZeroMemory(DataBuffer, DataTransferLength);
+
+    PLUN_LIST LunList;
+    LunList = DataBuffer;
+    LunList->LunListLength[3] = 0x8;
+    RtlCopyMemory(&LunList->Lun[0][0], &(Device->Bus), 1);
+    RtlCopyMemory(&LunList->Lun[0][1], &(Device->Lun), 1);
+
+    WNBD_LOG_INFO("Device bus: %u, Device lun: %u",
+        Device->Bus, Device->Lun);
+
+    SrbSetDataTransferLength(Srb, sizeof(PLUN_LIST));
```

Comment:
The data transfer length is incorrect, it should be sizeof(LUN_LIST) plus the size of the list entry.